neoantigenPredictor.wdl

```diff
@@ -221,20 +221,28 @@ task extractHLAs{
           ## limit HLA codes to Field 1 and 2, removing everything after (split on :, then join the first two fields)
           allele1=":".join(fields[2].split(":")[0:2])
           allele2=":".join(fields[5].split(":")[0:2])
-          hlas.append([hla_gene,allele1])
-          hlas.append([hla_gene,allele2])
+          if allele1 != ".":
+              hlas.append([hla_gene, allele1])
+          if allele2 != ".":
+              hlas.append([hla_gene, allele2])
 
     elif caller == "optitype":
       ### A,B and C Alleles are all on one line in columns 2-7 (A1,A2,B1,B2,C1,C2)
       with open(file) as f:
         lines=f.readlines()
         fields=lines[1].replace("*","").split()
-        hlas.append(["A","HLA-" + fields[1]])
-        hlas.append(["A","HLA-" + fields[2]])
-        hlas.append(["B","HLA-" + fields[3]])
-        hlas.append(["B","HLA-" + fields[4]])
-        hlas.append(["C","HLA-" + fields[5]])
-        hlas.append(["C","HLA-" + fields[6]])
+        if fields[1] != ".":
```

Comment:
do we know that optitype ever omits a "."?
i think this was seen in T1K.

perhaps checking for an expected pattern makes more sense